rpi5.yaml

```diff
@@ -0,0 +1,207 @@
+        - [PREFERRED_VERSION_xen-tools, "4.19.0+git"]
+        - [PREFERRED_VERSION_u-boot, "2024.04"]
+
+        # RPI specific
+        - [ENABLE_UART, "1"]
+        - [RPI_USE_U_BOOT, "1"]
+        - [PREFERRED_PROVIDER_u-boot-default-script, "xt-rpi-u-boot-scr"]
+        - [UART_BOORLOADER, "1"]
+        - [KERNEL_IMAGETYPE_UBOOT, "Image"]
+        - [DISTRO_FEATURES:append, " xen"]
+
+      build_target: "%{XT_DOMD_IMAGE}"
+      layers:
+        - "../meta-virtualization"
+        - "../meta-raspberrypi"
+        - "../meta-selinux"
+        - "../meta-openembedded/meta-oe"
+        - "../meta-openembedded/meta-networking"
+        - "../meta-openembedded/meta-perl"
+        - "../meta-openembedded/meta-python"
+        - "../meta-openembedded/meta-filesystems"
+        - "../meta-xt-common/meta-xt-domx"
+        - "../meta-xt-rpi5"
+
+      target_images:
+        - "tmp/deploy/images/%{MACHINE}/xen-%{MACHINE}"
+        - "tmp/deploy/images/%{MACHINE}/xenpolicy-%{MACHINE}"
+        - "tmp/deploy/images/%{MACHINE}/%{XT_XEN_DTB_NAME}"
+        - "tmp/deploy/images/%{MACHINE}/%{XT_DOMD_DTB_NAME}"
+        - "tmp/deploy/images/%{MACHINE}/bcm2712-rpi-5-b.dtb"
+        - "tmp/deploy/images/%{MACHINE}/%{XT_DOMD_IMAGE}-%{MACHINE}.rootfs.ext4"
+        - "tmp/deploy/images/%{MACHINE}/Image.gz-%{MACHINE}.bin"
+        - "tmp/deploy/images/%{MACHINE}/Image-%{MACHINE}.bin"
+        - "tmp/deploy/images/%{MACHINE}/bootfiles/bootcode.bin"
+        - "tmp/deploy/images/%{MACHINE}/bootfiles/config.txt"
+        - "tmp/deploy/images/%{MACHINE}/bootfiles/cmdline.txt"
+        - "tmp/deploy/images/%{MACHINE}/bootfiles/fixup.dat"
+        - "tmp/deploy/images/%{MACHINE}/bootfiles/fixup_cd.dat"
+        - "tmp/deploy/images/%{MACHINE}/bootfiles/fixup_db.dat"
+        - "tmp/deploy/images/%{MACHINE}/bootfiles/fixup_x.dat"
+        - "tmp/deploy/images/%{MACHINE}/bootfiles/fixup4.dat"
+        - "tmp/deploy/images/%{MACHINE}/bootfiles/fixup4cd.dat"
+        - "tmp/deploy/images/%{MACHINE}/bootfiles/fixup4db.dat"
+        - "tmp/deploy/images/%{MACHINE}/bootfiles/fixup4x.dat"
+        - "tmp/deploy/images/%{MACHINE}/bootfiles/start.elf"
+        - "tmp/deploy/images/%{MACHINE}/bootfiles/start_cd.elf"
+        - "tmp/deploy/images/%{MACHINE}/bootfiles/start_db.elf"
+        - "tmp/deploy/images/%{MACHINE}/bootfiles/start_x.elf"
+        - "tmp/deploy/images/%{MACHINE}/bootfiles/start4.elf"
+        - "tmp/deploy/images/%{MACHINE}/bootfiles/start4db.elf"
+        - "tmp/deploy/images/%{MACHINE}/bootfiles/start4cd.elf"
+        - "tmp/deploy/images/%{MACHINE}/bootfiles/start4x.elf"
+
+images:
+  full:
+    type: gpt
+    desc: "Full RPI image"
+    sector_size: 512
+    partitions:
+      boot:
+        gpt_type: C12A7328-F81F-11D2-BA4B-00A0C93EC93B # EFI System partition
+        type: vfat
+        size: 512 MiB
+        items:
+          "bootcode.bin": "%{YOCTOS_WORK_DIR}/%{DOMD_BUILD_DIR}/tmp/deploy/images/%{MACHINE}/bootfiles/bootcode.bin"
```

Comment:
I have tried to build using
```
moulin rpi5.yaml
ninja full.img
```
The build passed OK, nut got error on during the creation of the image:
```
moulin.yaml_helpers.YAMLProcessingError: Can't find file 'yocto/build-domd/tmp/deploy/images/raspberrypi5/bootfiles/bootcode.bin'   in "rpi5.yaml", line 163, column 27
```

Reply:
It happens if it was partial build. Follows command will help:
```
bitbake -c clean rpi-bootfiles -c clean rpi-config -c clean rpi-cmdline
```

Reply:
We had a discussion with some investigation of the recipes. For now, the issue looks quite similar to the "strange" issue with Xen images on the R-Car platform. And may take significant time to resolve.
So, taking into account that this is the start of the platform, we agreed to have the following proposal, that should be agreed upon with reviewers:
1. Create an issue for the repo "Some files are not put into the deploy" with a detailed explanation.
2. Add the "Known issues"  section to the readme.md with the description of the observed issue and proposed workaround:
```
cd yocto
. poky/oe-init-build-env build-domd/
bitbake -c clean rpi-bootfiles -c clean rpi-config -c clean rpi-cmdline
```
open a new terminal and repeat the creation of the image.

@lorc What do you think about this proposal?